### PR TITLE
#59 solution and test

### PR DESCRIPTION
**Pull request by ColinX13**
implement #59

### CODE REVIEW

File: solutions/59.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+const solution = (x) => {
+  if(x<2){
+    return false;
+  }
+  for(i=x; i>=1; i--){
+    if(x % i === 0 && i !== x && i !== 1){
+      return false;
+    };
+    return true;
+};
```
Comment (songz) [inline solutions/59.js]: fix indentation

File: solutions/59.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+//Function that takes in a number, returns true if number is prime. False otherwise.
+const solution = (x) => {
+  if(x<2){
+    return false;
+  }
+  for(i=x; i>=1; i--){
```
Comment (songz) [inline solutions/59.js]: `for( let i = x ...)` Bad practice to assign value to i without declaring it first

File: solutions/59.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    if(x % i === 0 && i !== x && i !== 1){
+      return false;
+    };
+    return true;
```
Comment (songz) [inline solutions/59.js]: you are returning true it the first iteration. Whats the point of the for loop?

File: solutions/59.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+//Function that takes in a number, returns true if number is prime. False otherwise.
```
Comment (songz) [inline solutions/59.js]: add comments to describe your function like this guy: https://github.com/llipio/algorithms/pull/55/files

File: test/59.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  });
+  it('-1 should not be a prime number', () => {
+    expect(solution(-1)).eql(false);
+  });
+
+  it('0 should not be a prime number', () => {
+    expect(solution(0)).eql(false);
+  });
+  it('14 should not be a prime number', () => {
+  expect(solution(15)).eql(false);
```
Comment (yjlim5) [inline test/59.js]: 15 --> 14

File: test/59.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  it('1 should not be a prime number', () => {
+    expect(solution(1)).eql(false);
+  });
+  it('13 is a prime number', () => {
+    expect(solution(13)).eql(true);
+  });
+  it('-1 should not be a prime number', () => {
+    expect(solution(-1)).eql(false);
+  });
+
```
Comment (yjlim5) [inline test/59.js]: remove empty line, for formatting consistency

File: solutions/59.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+// Solution by Colin Xie @ColinX13
+
+const solution = (x) => {
+  if(x<2){
+    return false;
+  }
+  for(let i=x-1; i>=2; i--){
+    if(x % i === 0 ){
+      return false;
+    };
```
Comment (yjlim5) [inline solutions/59.js]: Please remove ; in lines 17-18.  We don't need ; after for/if loops.

File: solutions/59.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
+//Takes a number x.
+//Returns false if x<2.
```
Comment (yjlim5) [inline solutions/59.js]: Please revise the header comments so that even a non-programmer can understand it.

**Comment by songz** [comment]
https://github.com/llipio/algorithms/pull/55/files

File: solutions/59.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
+//Input a number x into the 'solution' function.
+//Returns false if x<2.
+//Returns false if x is divisible by number not 1 and x when x>2.
```
Comment (songz) [inline solutions/59.js]: better indenting: https://github.com/llipio/algorithms/pull/55/files

File: solutions/59.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+//Otherwise return true.
+//input [1,13,-1,0,14]
+//output [false,true,false,false,false]
+
+// Solution by Colin Xie @ColinX13
+
+const solution = (x) => {
```
Comment (songz) [inline solutions/59.js]: better variable naming. instead of x, call it `prime`

File: solutions/59.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+  if(x<2){
+    return false;
+  }
+  for(let i=x-1; i>=2; i--){
```
Comment (songz) [inline solutions/59.js]: please pick better variable naming than `i`. perhaps `factor`?

File: test/59.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+    expect(solution(13)).eql(true);
+  });
+  it('-1 should not be a prime number', () => {
+    expect(solution(-1)).eql(false);
+  });
+  it('0 should not be a prime number', () => {
+    expect(solution(0)).eql(false);
+  });
+  it('15 should not be a prime number', () => {
+  expect(solution(15)).eql(false);
```
Comment (songz) [inline test/59.js]: indent please

File: solutions/59.js [hunk 13]
```diff
@@ -0,0 +1,25 @@
+// Finding prime numbers.
+// Takes in a number, returns true if the number is prime, false otherwise.
+// input: 1
+// output: false
+
+// Solution by Colin Xie @ColinX13
+/*
```
Comment (songz) [inline solutions/59.js]: `/**`

File: solutions/59.js [hunk 14]
```diff
@@ -0,0 +1,25 @@
+// Finding prime numbers.
```
Comment (songz) [inline solutions/59.js]: this whole section is unnecessary.

**Pull request merged**